lib/phoenix_live_view/tag_engine.ex

```diff
@@ -912,7 +912,15 @@ defmodule Phoenix.LiveView.TagEngine do
   defp handle_ast_attrs(state, attrs, tag_open_meta) do
     Enum.reduce(attrs, state, fn
       {name, value}, state when is_binary(value) ->
-        update_subengine(state, :handle_text, [[], ~s( #{name}="#{value}")])
+        try do
+          Phoenix.Component.MacroComponent.encode_binary_attribute(name, value)
+        rescue
+          e in ArgumentError ->
+            raise_syntax_error!(Exception.message(e), tag_open_meta, state)
+        else
+          attr ->
+            update_subengine(state, :handle_text, [[], attr])
+        end
```

Comment:
we could also just keep the ArgumentError, since the ParseError would not necessarily point to the exact attribute

Reply:
I am fine with letting the original one going through. It would most likely be a bug in the macro component, right?